.clang-tidy

```diff
@@ -46,6 +46,7 @@ Checks:
   -readability-function-cognitive-complexity, -readability-else-after-return,
   -readability-identifier-length, -readability-implicit-bool-conversion,
   -readability-magic-numbers, -readability-make-member-function-const,
+  -readability-redundant-member-init,
```

Comment:
Similar to the other PR, to document why we're doing this, can you add a note above like:

```
  # - readability-redundant-member-init hinders us from setting initializers to
  #   silence -Wmissing-designated-field-initializers.
```

Reply:
Yep, done and I rebased on top of #4540, so it should land first.